KPI/views/firefox_desktop_usage_forecast.view.lkml

```diff
@@ -0,0 +1,301 @@
+      {% condition firefox_desktop_usage_2021.content %} content {% endcondition %}
+      {% condition firefox_desktop_usage_2021.country %} country {% endcondition %}
+      {% condition firefox_desktop_usage_2021.distribution_id %} distribution_id {% endcondition %}
+      {% condition firefox_desktop_usage_2021.id_bucket %} id_bucket {% endcondition %}
+      {% condition firefox_desktop_usage_2021.medium %} medium {% endcondition %}
+      {% condition firefox_desktop_usage_2021.os %} os {% endcondition %}
+      {% condition firefox_desktop_usage_2021.source %} source {% endcondition %}
+      {% condition firefox_desktop_usage_2021.attributed %} (attributed) {% endcondition %}
+      """, r"(\(.*\))"), '');
+
+      DECLARE row_count INT64;
+
+      SET row_count = (
+        SELECT COUNT(*)
+        FROM mozdata.analysis.dou_forecasts
+        WHERE key = current_key);
+
+      IF row_count = 0 THEN
+
+      INSERT INTO mozdata.analysis.dou_forecasts_cache
+      WITH historic_data AS (
+        SELECT
+          CAST(submission_date AS TIMESTAMP) AS submission_date,
+
+          CAST(dau AS FLOAT64) AS dau_forecast,
+          CAST(NULL AS FLOAT64) AS dau_target,
+          CAST(NULL AS FLOAT64) AS dau_forecast_lower,
+          CAST(NULL AS FLOAT64) AS dau_forecast_upper,
+
+          CAST(new_profiles AS FLOAT64) AS new_profiles_forecast,
+          CAST(NULL AS FLOAT64) AS new_profiles_target,
+          CAST(NULL AS FLOAT64) AS new_profiles_forecast_lower,
+          CAST(NULL AS FLOAT64) AS new_profiles_forecast_upper,
+        FROM
+          ${firefox_desktop_usage_2021.SQL_TABLE_NAME}
+        WHERE
+          submission_date BETWEEN '2021-01-01' AND '2021-01-18'
+      ), predicted_data AS (
+        SELECT
+          forecast_timestamp AS submission_date,
+
+          dau.forecast_value AS dau_forecast,
+          dau.forecast_value * 1.05 AS dau_target,
+          dau.prediction_interval_lower_bound AS dau_forecast_lower,
+          dau.prediction_interval_upper_bound AS dau_forecast_upper,
+
+          np.forecast_value AS new_profiles_forecast,
+          np.forecast_value * 1.05 AS new_profiles_target,
+          np.prediction_interval_lower_bound AS new_profiles_forecast_lower,
+          np.prediction_interval_upper_bound AS new_profiles_forecast_upper,
+        FROM
+          ML.FORECAST(MODEL ${dau_model.SQL_TABLE_NAME},
+            STRUCT(365 AS horizon,
+            0.8 AS confidence_level)) AS dau
+        JOIN
+          ML.FORECAST(MODEL ${new_profiles_model.SQL_TABLE_NAME},
+            STRUCT(365 AS horizon,
+            0.8 AS confidence_level)) AS np
+          USING (forecast_timestamp)
+      )
+      SELECT
+        current_key AS key,
+        submission_date,
+
+        dau_forecast,
+        dau_target,
+        SUM(dau_forecast) OVER (ORDER BY submission_date ASC) AS cdou_forecast,
+        (SUM(dau_forecast) OVER (ORDER BY submission_date ASC)) * 1.05 AS cdou_target,
```

Comment:
I wonder if we should remove the targets here in order to avoid confusion. The +5% target is for the overall timeseries and is based on the "official" forecast only. We don't technically have a target for individual DAU segments.

Reply:
Also this made me realize that I have misunderstood how this was working when we talked today. Is there any way for us to display the "official" forecasts & target for the top-level unfiltered timeseries, and show the arima models otherwise? These are the targets forecasts that everyone has signed off on and have been tracking in the datastudio dashboard, and we should stick to them when displaying the overall DAU and new profiles time series. Or maybe I'm confused about what's going on here.

Reply:
Don't worry, it is confusing. We are using the official forecasts / targets for the top-level KPIs. There are a table and associated view, the table has the cache (with the on-the-fly modeling) and the view unions that with the official targets:

```
CREATE OR REPLACE TABLE
      mozdata.analysis.dou_forecasts_cache
CLUSTER BY key
    AS
      SELECT
        CAST(NULL AS STRING) AS key,
        CAST(NULL AS TIMESTAMP) AS submission_date,
        
        -- DAU
        CAST(NULL AS FLOAT64) AS dau_forecast,
        CAST(NULL AS FLOAT64) * 1.05 AS dau_target,
        CAST(NULL AS FLOAT64) AS cdou_forecast,
        CAST(NULL AS FLOAT64) AS cdou_target,
        CAST(NULL AS FLOAT64) AS dau_forecast_lower,
        CAST(NULL AS FLOAT64) AS dau_forecast_upper,
        
        -- New Profiles
        CAST(NULL AS FLOAT64) AS new_profiles_forecast,
        CAST(NULL AS FLOAT64) * 1.05 AS new_profiles_target,
        CAST(NULL AS FLOAT64) AS cum_new_profiles_forecast,
        CAST(NULL AS FLOAT64) AS cum_new_profiles_target,
        CAST(NULL AS FLOAT64) AS new_profiles_forecast_lower,
        CAST(NULL AS FLOAT64) AS new_profiles_forecast_upper,
      FROM
        mozdata.telemetry.clients_daily
      WHERE
        FALSE;

CREATE OR REPLACE VIEW
mozdata.analysis.dou_forecasts
AS
SELECT *
FROM mozdata.analysis.dou_forecasts_cache
UNION ALL
SELECT
  '' AS key,
  CAST(ds AS TIMESTAMP) AS submission_date,
  
  -- DAU
  dau_forecast.dau_forecast,
  dau_forecast.dau_forecast * 1.05 AS dau_target,
  SUM(dau_forecast.dau_forecast) OVER (PARTITION BY date_trunc(ds, YEAR) ORDER BY ds ASC) AS cdou_forecast,
  SUM(dau_forecast.dau_forecast) OVER (PARTITION BY date_trunc(ds, YEAR) ORDER BY ds ASC) * 1.05 AS cdou_target,
  dau_forecast.dau_forecast_lower,
  dau_forecast.dau_forecast_upper,
  
  -- New Profiles
  np_forecast.yhat AS new_profiles_forecast,
  np_forecast.yhat * 1.05 AS new_profiles_target,
  SUM(np_forecast.yhat) OVER (PARTITION BY date_trunc(ds, YEAR) ORDER BY ds ASC) AS cum_new_profiles_forecast,
  SUM(np_forecast.yhat) OVER (PARTITION BY date_trunc(ds, YEAR) ORDER BY ds ASC) * 1.05 AS cum_new_profiles_target,
  np_forecast.yhat_lower AS new_profiles_forecast_lower,
  np_forecast.yhat_upper AS new_profiles_forecast_upper,
FROM
  `mozdata.analysis.loines_desktop_dau_forecast_2021-01-19` AS dau_forecast
JOIN
  `mozdata.analysis.loines_desktop_new_profiles_forecast_2021-01-19` AS np_forecast
  ON dau_forecast.ds = DATE(np_forecast.date);
```

Reply:
I need to submit the above to `bigquery_etl` so it is centrally defined.